pmlc/dialect/pxa/ir/ops.cc

```diff
@@ -126,6 +136,22 @@ struct SimplifyDeadReduce : public OpRewritePattern<AffineReduceOp> {
   }
 };
 
+/// Fold vector_reduce operations with no uses. Vector_reduce has side effects
+/// on the heap, but can still be deleted if it has zero uses.
+struct SimplifyDeadVectorReduce
```

Comment:
I think we can reuse the existing `SimplifyDeadReduce`, the only difference is the template parameter.

Reply:
I agree, you should template SimplyDeadReduce on the operation type to and then just instantiate two versions in the patterns to reduce code duplication.

Reply:
It is done.